### PR TITLE
filtering using network_key

### PR DESCRIPTION
**Pull request by lszetg**
## Motivation and Context

- transactions are now filtered using the active network (its network_key)
- if a transaction has a status `Confirmed`, we don't fetch it from the crypto network for synchronization (nothing will change in it anyway)

## Summary

Provide a short summary of changes in this pull request.

- Change 1
- Change 2

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

**Comment by antbern** [comment]
@lszetg I remember we discussed this when creating the ticket, but instead of filtering on the currently selected network, we could also simply _provide the `network_key` as part of the transaction details and the filtering can be done on the client / integrators side_. Just FYI 💯

**Comment by lszetg** [comment]
> @lszetg I remember we discussed this when creating the ticket, but instead of filtering on the currently selected network, we could also simply _provide the `network_key` as part of the transaction details and the filtering can be done on the client / integrators side_. Just FYI 💯

the role of `get_wallet_tx_list` is also to update transactions. For IOTA we get the latest transactions from the network (well... when it still existed :D), for ETH we use this opportunity to synchronize (but only for the requested tx (to avoid overloading and do not hang there forever)). 

So... we're using
```rust
try_get_active_user_wallet() {
    ...
    let network = self.active_network.clone().ok_or(crate::Error::MissingNetwork)?;
     
   let wallet = active_user
        .wallet_manager
        .try_get(config, &self.access_token, repo, network, pin)
        .await?;
   Ok(wallet)
}
```

and then
```rust
let synchronized_transaction = wallet.get_wallet_tx(&transaction.transaction_id).await;
    -> wallet_evm.rs -> get_wallet_tx
```

to query the `current` network to get tx.

But we could refactor it. What do you think?

File: sdk/src/testing_utils.rs [hunk 1]
```diff
@@ -358,7 +358,7 @@ pub fn example_wallet_tx_info() -> WalletTxInfo {
         receiver: String::new(),
         incoming: true,
         amount: 20.0,
-        network_key: "some network".to_string(),
+        network_key: "IOTA".to_string(),
         status: "Complete".to_string(),
```
Comment (antbern) [inline sdk/src/testing_utils.rs:362]: Could we not make this status into a real enum instead of using a debug printed value from `iota-sdk` ? Could just be two states for now: `Pending` and `Complete` or similar.

Comment (lszetg) [reply sdk/src/testing_utils.rs:362]: aaah yeah, first of all it should be InclusionState to keep it consistent, i'll update it

maybe we could introduce new enum while integrating new iota?

Comment (antbern) [reply sdk/src/testing_utils.rs:362]: > maybe we could introduce new enum while integrating new iota?

Sure if you do not want to do it now we can always do it later :rocket

**Review by antbern — APPROVED** [review]

**Review by hauju — APPROVED** [review]
